### PR TITLE
Extract example package testing code to new example package

### PR DESCRIPTION
**Pull request by mauritsvanrees**
This contains three commits, and reading the individual commits may be easier to follow.

I first extract some testing code which is shared by the example packages into a new example package. The effect is that the example packages do not themselves import any plone.autoinclude code.

I use this in the second commit to test a few of our packages in a tox setup without `plone.autoinclude` but with `z3c.autoinclude`.  This will not work for all package, but only for the ones that actually have the standard `z3c.autoinclude.plugin` entry point with target plone in their `setup.py`.  In `tox.ini` and `buildoutz3c.cfg` I make sure to include only those test packages.  The tox environments `z3c-pip` and `z3c-buildout` are not run by default, because it is fine for us if they fail. But actually they both pass after the second commit.

In the third commit I add `example.z3cintegration`, which is an edited copy of `example.ploneintegration`, with basically a copy of the relevant zcml parts of current `Products.CMPlone` to include `z3c.autoinclude` and load all its plugins. With this, the z3c tox envs actually both fail. This may very well be because I set it up wrongly.

I had kind-of hoped that the new z3c tests would pass with buildout and fail with pip. That was one of the reasons that I wanted to work on this new package: reportedly `z3c.autoinclude` does not work with pip because it does not detect all zcml files. But maybe the reports only meant that `includeDependencies` was not working. That is an option in `z3c.autoinclude` which can be switched off, and we should probably do that in Plone 6 by default.

@tschorr Please see if these changes (or parts of them) make any sense to you.

### CODE REVIEW

**Review by tschorr — APPROVED** [review]
Lgtm. I like the general idea of this, let's see where this gets us.

File: src/plone/autoinclude/tests/utils.py [hunk 1]
```diff
@@ -7,6 +7,9 @@ def get_configuration_context(package=None):
 
     Various functions take a configuration context as argument.
     From looking at zope.configuration.xmlconfig.file the following seems about right.
+
+    Note: there is a copy of this function in example.basetestpackage.
+    The duplication is deliberate: I don't one to to import code from the other, for now.
```
[the remark this answers is not in the document]
Comment (mauritsvanrees) [reply src/plone/autoinclude/tests/utils.py:12]: Comment fixed.

File: test-packages/example.basetestpackage/src/example/basetestpackage/utils.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+def get_configuration_context(package=None):
+    """Get configuration context.
+
+    Various functions take a configuration context as argument.
+    From looking at zope.configuration.xmlconfig.file the following seems about right.
+
+    Note: this is a copy of a function in plone.autoinclude.tests.utils.
+    The duplication is deliberate: I don't one to import code from the other, for now.
```
[the remark this answers is not in the document]
Comment (mauritsvanrees) [reply test-packages/example.basetestpackage/src/example/basetestpackage/utils.py:12]: Comment fixed.

**Comment by mauritsvanrees** [comment]
Thanks for the review. I have fixed the comments locally and then merged to main.

**Pull request merged**